sklearn/linear_model/_ridge.py

```diff
@@ -41,17 +41,28 @@
 
 
 def _solve_sparse_cg(
-    X, y, alpha, max_iter=None, tol=1e-3, verbose=0, X_offset=None, X_scale=None
+    X,
+    y,
+    alpha,
+    max_iter=None,
+    tol=1e-3,
+    verbose=0,
+    X_offset=None,
+    X_scale=None,
+    sample_weight_sqrt=None,
 ):
+    if sample_weight_sqrt is None:
+        sample_weight_sqrt = np.ones(X.shape[0], dtype=X.dtype)
+
     def _get_rescaled_operator(X):
 
         X_offset_scale = X_offset / X_scale
 
         def matvec(b):
-            return X.dot(b) - b.dot(X_offset_scale)
+            return X.dot(b) - sample_weight_sqrt * b.dot(X_offset_scale)
 
         def rmatvec(b):
-            return X.T.dot(b) - X_offset_scale * np.sum(b)
+            return X.T.dot(b) - X_offset_scale * b.dot(sample_weight_sqrt)
```

Comment:
Does this slow down the case with no sample weights? An that case, we might consider a case distinction.
I guess it is negligible.

Reply:
The first one is 1 scalar multiplication vs a sparse matrix - vector mult and a dot product.
The second one is a dot product instead of a sum which means n_features more multiplications vs a sparse matrix - vector mult.
I think it's pretty negligible